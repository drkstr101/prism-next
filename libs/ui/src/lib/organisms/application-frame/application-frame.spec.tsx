import {
  Button,
  ButtonGroup,
  Checkbox,
  Content,
  Divider,
  Flex,
  Footer,
  Form,
  Header,
  Heading,
  Link,
  Provider,
  Text,
  TextField,
  defaultTheme,
} from '@adobe/react-spectrum';
import Book from '@spectrum-icons/workflow/Book';
import { render } from '@testing-library/react';

import ApplicationFrame from './application-frame';

const ExampleContent = () => (
  <>
    <Heading>
      <Flex alignItems="center" gap="size-100">
        <Book size="S" />
        <Text>Register for newsletter</Text>
      </Flex>
    </Heading>
    <Header>
      <Link>
        <a href="//example.com" target="_blank" rel="noreferrer">
          What is this?
        </a>
      </Link>
    </Header>
    <Divider />
    <Content>
      <Form>
        <TextField label="First Name" autoFocus />
        <TextField label="Last Name" />
        <TextField label="Street Address" />
        <TextField label="City" />
      </Form>
    </Content>
    <Footer>
      <Checkbox>I want to receive updates for exclusive offers in my area.</Checkbox>
    </Footer>
    <ButtonGroup>
      <Button variant="secondary">Cancel</Button>
      <Button variant="accent">Register</Button>
    </ButtonGroup>
  </>
);

describe('ApplicationFrame', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Provider theme={defaultTheme}>
        <ApplicationFrame>
          <ExampleContent />
        </ApplicationFrame>
      </Provider>
    );
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
});
