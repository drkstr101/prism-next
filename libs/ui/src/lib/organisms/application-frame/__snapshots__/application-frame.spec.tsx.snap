// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationFrame should render successfully 1`] = `
<div>
  <div
    class="JuTe6q_spectrum _5QszkG_spectrum _5QszkG_i18nFontFamily PFjRbG_spectrum--light HAZavG_spectrum--large zA6MfG_spectrum zA6MfG_spectrum--dark zA6MfG_spectrum--darkest zA6MfG_spectrum--large zA6MfG_spectrum--light zA6MfG_spectrum--lightest zA6MfG_spectrum--medium"
    dir="ltr"
    lang="en-US"
    style="isolation: isolate; color-scheme: light;"
  >
    <div
      style="grid-template-areas: "header"
"nav"
"content"
"footer"; grid-template-columns: var(--spectrum-global-dimension-undefined, var(--spectrum-alias-undefined)); gap: var(--spectrum-global-dimension-size-50, var(--spectrum-alias-size-50)); display: grid;"
    >
      <div
        style="grid-area: header;"
      />
      <div
        style="grid-area: nav;"
      >
        <div
          class="NW91UW_flex-container"
        >
          <div
            class="NW91UW_flex NW91UW_flex-gap"
            style="flex-direction: row; --gap: var(--spectrum-global-dimension-size-100, var(--spectrum-alias-size-100));"
          >
            <div
              style="min-width: var(--spectrum-global-dimension-size-900, var(--spectrum-alias-size-900));"
            />
            <div
              style="min-width: var(--spectrum-global-dimension-size-900, var(--spectrum-alias-size-900));"
            />
            <div
              style="min-width: var(--spectrum-global-dimension-size-900, var(--spectrum-alias-size-900));"
            />
          </div>
        </div>
      </div>
      <div
        style="grid-area: content;"
      >
        <h3>
          <div
            class="NW91UW_flex-container"
          >
            <div
              class="NW91UW_flex NW91UW_flex-gap"
              style="align-items: center; --gap: var(--spectrum-global-dimension-size-100, var(--spectrum-alias-size-100));"
            >
              <svg
                aria-hidden="true"
                class="yxBNXG_spectrum-Icon yxBNXG_spectrum-Icon--sizeS"
                focusable="false"
                role="img"
                viewBox="0 0 36 36"
              >
                <path
                  d="M19.782,28H9.995a4,4,0,0,1,0-8H20.518a1,1,0,0,0,.8-.4l11.1-14.8a.5.5,0,0,0-.4-.8H16.025a1,1,0,0,0-.8.4L3.522,19.328A7.981,7.981,0,0,0,9.969,32l10.549,0a1,1,0,0,0,.8-.4l11.1-14.8a.5.5,0,0,0-.4-.8H28.782Z"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Register for newsletter
              </span>
            </div>
          </div>
        </h3>
        <header>
          <a
            class="WBgRPa_spectrum-Link spectrum-Link--primary"
            href="//example.com"
            rel="noreferrer"
            target="_blank"
          >
            What is this?
          </a>
        </header>
        <hr
          class="cTbPrq_spectrum-Rule cTbPrq_spectrum-Rule--large cTbPrq_spectrum-Rule--horizontal"
        />
        <section>
          <form
            class="A-HlBa_spectrum-Form A-HlBa_spectrum-Form--positionTop"
            novalidate=""
          >
            <div
              class="A-HlBa_spectrum-Field A-HlBa_spectrum-Field--positionTop YO3Nla_spectrum-Textfield-wrapper"
            >
              <label
                class="A-HlBa_spectrum-FieldLabel"
                for="react-aria-:r0:"
                id="react-aria-:r1:"
              >
                First Name
              </label>
              <div
                class="YO3Nla_spectrum-Textfield YO3Nla_spectrum-FocusRing YO3Nla_spectrum-FocusRing-ring YO3Nla_focus-ring A-HlBa_spectrum-Field-field"
              >
                <input
                  aria-labelledby="react-aria-:r1:"
                  class="YO3Nla_spectrum-Textfield-input YO3Nla_i18nFontFamily"
                  id="react-aria-:r0:"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="A-HlBa_spectrum-Field A-HlBa_spectrum-Field--positionTop YO3Nla_spectrum-Textfield-wrapper"
            >
              <label
                class="A-HlBa_spectrum-FieldLabel"
                for="react-aria-:r7:"
                id="react-aria-:r8:"
              >
                Last Name
              </label>
              <div
                class="YO3Nla_spectrum-Textfield YO3Nla_spectrum-FocusRing YO3Nla_spectrum-FocusRing-ring A-HlBa_spectrum-Field-field"
              >
                <input
                  aria-labelledby="react-aria-:r8:"
                  class="YO3Nla_spectrum-Textfield-input YO3Nla_i18nFontFamily"
                  id="react-aria-:r7:"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="A-HlBa_spectrum-Field A-HlBa_spectrum-Field--positionTop YO3Nla_spectrum-Textfield-wrapper"
            >
              <label
                class="A-HlBa_spectrum-FieldLabel"
                for="react-aria-:re:"
                id="react-aria-:rf:"
              >
                Street Address
              </label>
              <div
                class="YO3Nla_spectrum-Textfield YO3Nla_spectrum-FocusRing YO3Nla_spectrum-FocusRing-ring A-HlBa_spectrum-Field-field"
              >
                <input
                  aria-labelledby="react-aria-:rf:"
                  class="YO3Nla_spectrum-Textfield-input YO3Nla_i18nFontFamily"
                  id="react-aria-:re:"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="A-HlBa_spectrum-Field A-HlBa_spectrum-Field--positionTop YO3Nla_spectrum-Textfield-wrapper"
            >
              <label
                class="A-HlBa_spectrum-FieldLabel"
                for="react-aria-:rl:"
                id="react-aria-:rm:"
              >
                City
              </label>
              <div
                class="YO3Nla_spectrum-Textfield YO3Nla_spectrum-FocusRing YO3Nla_spectrum-FocusRing-ring A-HlBa_spectrum-Field-field"
              >
                <input
                  aria-labelledby="react-aria-:rm:"
                  class="YO3Nla_spectrum-Textfield-input YO3Nla_i18nFontFamily"
                  id="react-aria-:rl:"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </section>
        <footer>
          <label
            class="ISsn1a_spectrum-Checkbox ISsn1a_spectrum-Checkbox--quiet"
          >
            <input
              class="ISsn1a_spectrum-Checkbox-input"
              type="checkbox"
            />
            <span
              class="ISsn1a_spectrum-Checkbox-box"
            >
              <svg
                aria-hidden="true"
                class="yxBNXG_spectrum-Icon yxBNXG_spectrum-UIIcon-CheckmarkSmall ISsn1a_spectrum-Checkbox-checkmark"
                focusable="false"
                role="img"
              >
                <path
                  d="M4.5 11a.999.999 0 0 1-.788-.385l-3-4a1 1 0 1 1 1.576-1.23L4.5 8.376l5.212-6.99a1 1 0 1 1 1.576 1.23l-6 8A.999.999 0 0 1 4.5 11z"
                />
              </svg>
            </span>
            <span
              class="ISsn1a_spectrum-Checkbox-label"
            >
              I want to receive updates for exclusive offers in my area.
            </span>
          </label>
        </footer>
        <div
          class="aaz5ma_spectrum-ButtonGroup"
        >
          <button
            class="Dniwja_spectrum-Button Dniwja_spectrum-BaseButton Dniwja_i18nFontFamily Dniwja_spectrum-FocusRing Dniwja_spectrum-FocusRing-ring aaz5ma_spectrum-ButtonGroup-Button"
            data-style="outline"
            data-variant="secondary"
            type="button"
          >
            <span
              class="Dniwja_spectrum-Button-label"
            >
              Cancel
            </span>
          </button>
          <button
            class="Dniwja_spectrum-Button Dniwja_spectrum-BaseButton Dniwja_i18nFontFamily Dniwja_spectrum-FocusRing Dniwja_spectrum-FocusRing-ring aaz5ma_spectrum-ButtonGroup-Button"
            data-style="fill"
            data-variant="accent"
            type="button"
          >
            <span
              class="Dniwja_spectrum-Button-label"
            >
              Register
            </span>
          </button>
        </div>
      </div>
      <div
        hidden=""
        style="grid-area: panel; min-height: var(--spectrum-global-dimension-size-1000, var(--spectrum-alias-size-1000)); display: none;"
      />
      <div
        style="grid-area: footer;"
      />
    </div>
  </div>
</div>
`;
